RevenueCatUI/CustomerCenter/Views/Virtual Currencies/VirtualCurrenciesListSection.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  VirtualCurrenciesListSection.swift
+//
+//  Created by Will Taylor on 4/22/25.
+
+#if os(iOS)
+
+import Foundation
+import RevenueCat
+import SwiftUI
+
+@available(iOS 15.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+/// A SwiftUI view that displays a list of virtual currency balances in a section format.
+///
+/// This view shows up to three virtual currencies sorted by balance in descending order.
+/// If there are more than three currencies, a "See All" button is displayed that navigates
+/// to a full list of virtual currencies.
+struct VirtualCurrenciesListSection: View {
+
+    @Environment(\.localization)
+    private var localization: CustomerCenterConfigData.Localization
+
+    @Environment(\.navigationOptions)
+    var navigationOptions
+
+    @State private var showVirtualCurrenciesListScreen = false
+
+    private let virtualCurrencies: [VirtualCurrencyBalanceListRow.RowData]
+    private let displayShowAllButton: Bool
+    private let purchasesProvider: CustomerCenterPurchasesType
+
+    init(
+        virtualCurrencies: [String: RevenueCat.VirtualCurrencyInfo],
+        purchasesProvider: CustomerCenterPurchasesType
+    ) {
+        let sortedCurrencies = virtualCurrencies.map {
+            VirtualCurrencyBalanceListRow.RowData(
+                virtualCurrencyCode: $0.key,
+                balance: $0.value.balance
+            )
+        }
+            .sorted(by: { $0.balance > $1.balance })
```

Comment:
I'm not sure about this sorting. I mean, it makes sense for now, but I could see some customers wanting their own sorting order logic. Is this something that would make sense to have defined in the dashboard? (I'm thinking, for example, there's 1 VC that is much more valuable than any amount of any other VC)

Reply:
I think that's a great idea! It feels a little out of scope for now, but it's definitely something that we could add if we get customer requests for additional sorting methods!